### PR TITLE
[Code health] Use form definition when deserializing observations from remote db

### PR DESCRIPTION
**Pull request by gino-m**
Contributes to #754. This will be used to allow date, time, and other Ground form field types to use the same remote and local representation as other field types.

@atalekar @HemanParbhakar PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java [hunk 1]
```diff
@@ -54,32 +56,61 @@ static Observation toObservation(Feature feature, DocumentSnapshot snapshot)
...
+      try {
+        Object obj = docResponses.get(fieldId);
+        putResponse(fieldId, form, obj, responses);
+      } catch (DataStoreException e) {
+        Timber.d("Field " + fieldId + "in remote db in observation " + observationId + ": " + e);
       }
     }
     return responses.build();
   }
+
```
Comment (atalekar) [inline gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java:81]: many of these classes are too low level to not have 100% unit test coverage. LGTM in general but strongly recommend having good unit test coverage here. In future across release if there are changes in this section, one will need to be quite a bit careful around backward compatibility etc.

**Review by atalekar — DISMISSED** [review]

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java (hunk 1 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java:81]: @atalekar Very good points. We started out working under the assumption that everything we're building is a throw-away prototype/demo, so we started without test coverage. The initial assumption probably no longer holds, but our 20% team is so small that adding tests retroactively wasn't feasible. That said, I agree we probably shouldn't kick the can further down the road. I'll try to add some tests for this class!

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java:81]: Tests still a WIP, but added basic skeleton to get started. I'll flesh them out tomorrow, but please feel free to take a look at the last few commits now if you have a cycle or two. Thanks!

Comment (atalekar) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ObservationConverter.java:81]: lgtm - thanks a lot!

**Review by gino-m — COMMENTED** [review]
100% coverage of `ObservationConverter`. RFAL!

**Review by atalekar — APPROVED** [review]

**Comment by gino-m** [comment]
@atalekar Sorry I forgot to push the PMD error fixes, so your LGTM got invalidated. Could you kindly restamp?

**Review by atalekar — APPROVED** [review]

**Pull request merged**